src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -1134,14 +1134,20 @@ def create_nic(resource_group_name, network_interface_name, subnet, location=Non
                public_ip_address=None, virtual_network_name=None, enable_accelerated_networking=None):
     client = _network_client_factory().network_interfaces
     NetworkInterface = get_sdk(ResourceType.MGMT_NETWORK, 'NetworkInterface', mod='models')
+
+    network_interface_args = {}
+    if supported_api_version(ResourceType.MGMT_NETWORK, min_api='2016-09-01'):
+        network_interface_args['enable_accelerated_networking'] = enable_accelerated_networking
+
     NetworkInterfaceDnsSettings = get_sdk(
         ResourceType.MGMT_NETWORK,
         'NetworkInterfaceDnsSettings', mod='models')
 
     dns_settings = NetworkInterfaceDnsSettings(internal_dns_name_label=internal_dns_name_label,
                                                dns_servers=dns_servers or [])
+
     nic = NetworkInterface(location=location, tags=tags, enable_ip_forwarding=enable_ip_forwarding,
-                           dns_settings=dns_settings, enable_accelerated_networking=enable_accelerated_networking)
```

Comment:
This would be more concise and consistent with the rest of the CLI patterns:

```Python
nic = NetworkInterface(location=location, tags=tags, enable_ip_forwarding=enable_ip_forwarding, dns_settings=dns_settings)

if supported_api_version(ResourceType.MGMT_NETWORK, min_api='2016-09-01'):
  nic.enable_accelerated_networking = enable_accelerated_networking

``` 